src/js/component.js

```diff
@@ -220,28 +220,28 @@ class Component {
    * @return {string}
    *         The localized string or if no localization exists the english string.
    */
-  localize(string) {
+  localize(string, tokens) {
     const code = this.player_.language && this.player_.language();
     const languages = this.player_.languages && this.player_.languages();
+    const language = languages && languages[code];
+    const primaryCode = code && code.split('-')[0];
+    const primaryLang = languages && languages[primaryCode];
 
-    if (!code || !languages) {
-      return string;
-    }
-
-    const language = languages[code];
+    let localizedString = string;
 
     if (language && language[string]) {
-      return language[string];
+      localizedString = language[string];
+    } else if (primaryLang && primaryLang[string]) {
+      localizedString = primaryLang[string];
     }
 
-    const primaryCode = code.split('-')[0];
-    const primaryLang = languages[primaryCode];
-
-    if (primaryLang && primaryLang[string]) {
-      return primaryLang[string];
+    if (tokens) {
+      localizedString = localizedString.replace(/\{(\d+)\}/g, function(match, index) {
+        return tokens[index - 1];
```

Comment:
Another thing I can think of to help with this is having the translation string and then default be two different things. So, for this case I can do something like `this.localize('progress bar timing', [currentTime, duration], '{1} {2}')` and the en.json would have `progress bar timing: '{1} {2}'`. That way we have a default, that's described, but people who only look at the json would have a better idea about what's happening.
Not quite sure if there's the best approach but a thought.